addons/voice/org.openhab.voice.voicerss/src/main/java/org/openhab/voice/voicerss/internal/cloudapi/CachedVoiceRSSCloudImplementation.java

```diff
@@ -0,0 +1,132 @@
+/**
+ * Copyright (c) 2014-2016 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.voice.voicerss.internal.cloudapi;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.UnsupportedEncodingException;
+import java.math.BigInteger;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class implements a cache for the retrieved audio data. It will preserve
+ * them in file system, as .mp3 files with an additional .txt file to indidate
+ * what content is in the audio file.
+ * 
+ * @author Jochen Hiller - Initial contribution
+ */
+public class CachedVoiceRSSCloudImplementation extends VoiceRSSCloudImplementation {
+
+	private final Logger logger = LoggerFactory.getLogger(CachedVoiceRSSCloudImplementation.class);
+
+	private final File cacheFolder;
+
+	public CachedVoiceRSSCloudImplementation(String cacheFolderName) {
+		if (cacheFolderName == null) {
+			throw new RuntimeException("Folder for cache must be defined");
+		}
+		// Lazy create the cache folder
+		cacheFolder = new File(cacheFolderName);
+		if (!cacheFolder.exists()) {
+			cacheFolder.mkdirs();
+		}
+	}
+
+	public File getTextToSpeechAsFile(String apiKey, String text, String locale, String audioFormat)
+			throws IOException {
+		String fileNameInCache = getUniqeFilenameForText(text, locale);
+		// check if in cache
+		File audioFileInCache = new File(cacheFolder, fileNameInCache + ".mp3");
+		if (audioFileInCache.exists()) {
+			return audioFileInCache;
+		}
+
+		// if not in cache, get audio data and put to cache
+		InputStream is = super.getTextToSpeech(apiKey, text, locale, audioFormat);
+		FileOutputStream fos;
+		try {
+			fos = new FileOutputStream(audioFileInCache);
+			copyStream(is, fos);
```

Comment:
Instead of implementing it yourself, I would suggest to use Apache Commons IOUtils.
Or did you avoid using this lib on purpose, because you don't want to have dependencies for the command line use?